cli/daemon/engine/trace2/sqlite/write.go

```diff
@@ -177,49 +177,65 @@ func (s *Store) updateSpanStartIndex(ctx context.Context, meta *trace2.Meta, ev
 	isRoot := start.ParentSpanId == nil
 	if req := start.GetRequest(); req != nil {
 		extRequestID := req.RequestHeaders[http.CanonicalHeaderKey("X-Request-ID")]
+		var parentSpanID *string
+		if start.ParentSpanId != nil {
+			encodedParentSpanID := encodeSpanID(*start.ParentSpanId)
+			parentSpanID = &encodedParentSpanID
+		}
 		_, err := s.db.ExecContext(ctx, `
 			INSERT INTO trace_span_index (
-				app_id, trace_id, span_id, span_type, started_at, is_root, service_name, endpoint_name, external_request_id, has_response, test_skipped
-			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, false, false)
+				app_id, trace_id, span_id, span_type, started_at, is_root, service_name, endpoint_name, external_request_id, has_response, test_skipped, parent_span_id
+			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, false, false, ?)
 			ON CONFLICT (trace_id, span_id) DO UPDATE SET
 				is_root = excluded.is_root,
 				service_name = excluded.service_name,
 				endpoint_name = excluded.endpoint_name,
-				external_request_id = excluded.external_request_id
+				external_request_id = excluded.external_request_id,
+				parent_span_id = excluded.parent_span_id
 		`, meta.AppID, encodeTraceID(ev.TraceId), encodeSpanID(ev.SpanId),
 			tracepbcli.SpanSummary_REQUEST, ev.EventTime.AsTime().UnixNano(),
-			isRoot, req.ServiceName, req.EndpointName, extRequestID)
+			isRoot, req.ServiceName, req.EndpointName, extRequestID, parentSpanID)
 		if err != nil {
 			return errors.Wrap(err, "insert trace span event")
 		}
 		return nil
 	}
 
 	if auth := start.GetAuth(); auth != nil {
+		var parentSpanID *string
+		if start.ParentSpanId != nil {
+			encodedParentSpanID := encodeSpanID(*start.ParentSpanId)
+			parentSpanID = &encodedParentSpanID
+		}
 		_, err := s.db.ExecContext(ctx, `
 			INSERT INTO trace_span_index (
 				app_id, trace_id, span_id, span_type, started_at, is_root, service_name,
-				endpoint_name, has_response, test_skipped
-			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, false, false)
+				endpoint_name, has_response, test_skipped, parent_span_id
+			) VALUES (?, ?, ?, ?, ?, ?, ?, ?, false, false, ?)
```

Comment:
Nitpick: I generally would put the `parent_span_id` after `span_id` in the list, rather than at the end, since that makes more sense semantically. I would also try to avoid putting a placeholder after the two `false` constants.